src/android/src/main/java/io/callstack/react/fbads/RewardedVideoAdManager.java

```diff
@@ -0,0 +1,132 @@
+package io.callstack.react.fbads;
+
+import com.facebook.ads.Ad;
+import com.facebook.ads.AdError;
+import com.facebook.ads.RewardedVideoAd;
+import com.facebook.ads.RewardedVideoAdListener;
+import com.facebook.react.bridge.LifecycleEventListener;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+
+import android.util.Log;
+
+public class RewardedVideoAdManager extends ReactContextBaseJavaModule implements LifecycleEventListener {
+
+    private Promise mLoadedPromise = null;
+    private Promise mShowPromise = null;
+
+    private String TAG = "RewardedVideo";
+
+    private RewardedVideoAd mRewardedVideoAd;
+  
+
+    public RewardedVideoAdManager(ReactApplicationContext reactContext) {
+        super(reactContext);
+        reactContext.addLifecycleEventListener(this);
+    }
+
+    @ReactMethod
+    public void showAd(Promise p) {
+        if (mShowPromise != null) {
+            p.reject("E_FAILED_TO_SHOW", "Only one `showAd` can be called at once");
+            return;
+        }
+        if (mRewardedVideoAd != null && mRewardedVideoAd.isAdLoaded()) {
+            mRewardedVideoAd.show();
+            mShowPromise = p;
+        }
+    }
+
+    @ReactMethod
+    public void loadAd(String placementId, Promise p) {
+        if (mLoadedPromise != null) {
+            p.reject("E_FAILED_TO_SHOW", "Only one `loadAd` can be called at once");
+            return;
+        }
+
+        ReactApplicationContext reactContext = this.getReactApplicationContext();
+        mRewardedVideoAd = new RewardedVideoAd(reactContext, placementId);
```

Comment:
We should probably call destroy() on mRewardedVdeoAd before reassigning it to a new instance if it is not null. Currently it can provide to memory leaks I think. It is described in docs in step 2 - https://developers.facebook.com/docs/audience-network/android/rewarded-video

Reply:
Sounds good I'll add it.